sihl/src/database.ml

```diff
@@ -138,22 +150,42 @@ let fetch_pool () =
     |> Caqti_lwt.connect_pool ~max_size:pool_size
     |> (function
     | Ok pool ->
-      pool_ref := Some pool;
+      main_pool_ref := Some pool;
       pool
     | Error err ->
       let msg = "Failed to connect to DB pool" in
       Logs.err (fun m -> m "%s %s" msg (Caqti_error.show err));
       raise (Contract_database.Exception ("Failed to create pool " ^ msg)))
 ;;
 
+let add_pool ?(pool_size = 10) name database_url =
+  database_url
+  |> Uri.of_string
+  |> Caqti_lwt.connect_pool ~max_size:pool_size
+  |> function
+  | Ok pool ->
+    if Option.is_some (Hashtbl.find_opt pools name)
+    then (
+      let msg =
+        Format.sprintf "Connection pool with name '%s' exists already" name
+      in
+      Logs.err (fun m -> m "%s" msg);
+      raise (Contract_database.Exception ("Failed to create pool " ^ msg)))
```

Comment:
Please add a colon or dot after `Failed to create pool` message

Reply:
Fixed.